### PR TITLE
ready-for-final-review: ibu cnf: add upgrade abort test case

### PR DESCRIPTION
**Pull request by mpmaruthu**
adding upgrade abort test case

### CODE REVIEW

**Comment by mpmaruthu** [comment]
@mcornea , @klaskosk , Please review this PR when you get a chance. Thanks!

**Review by klaskosk — CHANGES_REQUESTED** [review]
Just a few comments about eco-goinfra, otherwise looks good

File: tests/lca/imagebasedupgrade/cnf/internal/cnfhelper/cnfhelper.go [hunk 1]
```diff
@@ -29,3 +40,86 @@ func DeleteIbuTestCguOnTargetHub(client *clients.Settings, name, namespace strin
...
+
+	// Check the returned policy list for the specific policy
+	for _, policy := range policyList.Items {
+		if policy.Name == policyName {
+			return policy, nil
+		}
+	}
+
+	return policiesv1.Policy{}, errors.New("resource not found")
+}
```
[the remark this answers is not in the document]
Comment (mpmaruthu) [reply tests/lca/imagebasedupgrade/cnf/internal/cnfhelper/cnfhelper.go]: Thanks. Required changes updated.

File: tests/lca/imagebasedupgrade/cnf/internal/cnfhelper/cnfhelper.go [hunk 2]
```diff
@@ -29,3 +40,86 @@ func DeleteIbuTestCguOnTargetHub(client *clients.Settings, name, namespace strin
...
+		return nil
+	}
+
+	// Check its compliance state
+	if policy.Status.ComplianceState == complianceState {
+		return nil
+	}
+
+	return err
+}
```
[the remark this answers is not in the document]
Comment (mpmaruthu) [reply tests/lca/imagebasedupgrade/cnf/internal/cnfhelper/cnfhelper.go]: Thanks. Required changes implemented.

**Review by klaskosk — APPROVED** [review]
/lgtm

**Comment by mcornea** [comment]
lgtm

**Review by mcornea — APPROVED** [review]

**Pull request merged**